spec/player_spec.rb

```diff
@@ -1,53 +1,55 @@
 require "spec_helper"
 require "tictactoe"
+require "fake_display"
 
 RSpec.describe Player do
-  class FakeDisplay
-    attr_reader :messages
-
-    def initialize(inputs: [])
-      @inputs = inputs
-      @messages = []
-    end
-
-    def output(message)
-      @messages << message
-      nil
-    end
-
-    def input
-      @inputs.shift
-    end
-  end
-
-  describe "#move" do
+  describe "#selection" do
     context "with a valid entry between 1 and 9" do
       let(:player) { Player.new(display) }
       let(:display) { FakeDisplay.new(inputs: ["5"]) }
+      let(:board) { Board.new }
 
       it "prompts the player for a move" do
-        player.selection
+        player.selection(board)
 
         expect(display.messages).to include(/Please select your move/)
       end
 
       it "returns the player's chosen space" do
-        expect(player.selection).to eq(5)
+        expect(player.selection(board)).to eq(5)
       end
     end
 
     context "with an invalid entry" do
       let(:player) { Player.new(display) }
       let(:display) { FakeDisplay.new(inputs: ["0", "?", "5"]) }
-
+      let(:board) { Board.new }
       it "prompts the player for a move until receiving a valid move" do
-        player.selection
+        player.selection(board)
 
         expect(display.messages).to contain_exactly(/Please select your move/, /Invalid entry./, /Please select your move/, /Invalid entry./, /Please select your move/)
       end
 
       it "returns the player's chosen space" do
-        expect(player.selection).to eq(5)
+        expect(player.selection(board)).to eq(5)
+      end
+    end
+
+    context "with an unavailable entry" do
+      let(:player) { Player.new(display) }
+      let(:display) { FakeDisplay.new(inputs: ["1", "1", "5"]) }
+      let(:board) { Board.new }
+
+      it "prompts the player for a move until receiving an available move" do
+        board.place_token(1, "X")
+        player.selection(board)
+
+        expect(display.messages).to contain_exactly(/Please select your move/, /Selection taken/, /Please select your move/, /Selection taken/, /Please select your move/)
```

Comment:
This is a pretty long line. My recommendation would be to aim for lines that are no longer than 120 characters.

Reply:
will adjust

Reply:
For context, I believe that limiting 120 characters will prevent awkward line breaks in the diff. It also will help ensure that all every character in the line will be viewable in your editor.